packages/core/src/cache/classes/Surface.ts

```diff
@@ -1,61 +1,146 @@
-import type { SurfaceData, Point3, ISurface, RGB } from '../../types';
+import type { SurfaceData, Point3, RGB } from '../../types';
 
-interface SurfaceProps {
-  id: string;
-  data: SurfaceData;
-  frameOfReferenceUID: string;
-  color?: Point3;
-}
+type SurfaceProps = SurfaceData;
 
 /**
  * Surface class for storing surface data
+ * Each Surface represents a single segment in a 3D volume.
  */
 export class Surface {
   readonly id: string;
   readonly sizeInBytes: number;
   readonly frameOfReferenceUID: string;
-  private color: RGB = [200, 0, 0]; // default color
-  private points: number[];
-  private polys: number[];
+  private _color: RGB = [200, 0, 0]; // default color
+  private _points: number[];
```

Comment:
The PointsManager manages the total points array/re-allocates as necessary and allows direct access as a Point3 reference value.